### ISSUE TITLE
Tarsnap stats get mangled with recovered checkpoints

### ISSUE THREAD

**Issue by cperciva** [Tarsnap/tarsnap#113]
There's a glitch in the handling of the "all archives" statistics relating to recovered checkpoints.  Specifically, if an archive is recovered from a checkpoint the "extra" statistics for the metadata and metaindex blocks aren't included; but they're deducted off when that recovered archive is deleted.  This shows up when users delete _all_ of their archives, and then get told that they have 18.4 exabytes of data stored (aka. a small negative number interpreted as an unsigned 64-bit value).

Some code in the multitape layer needs to be fixed -- I think in creating checkpoints we'll need to increment the "extra" statistics before writing out the directory file and then adjust back.

**Comment by gperciva** [comment]
Note to self: this can be illustrated without "recovered" or "deleted":

- (set up `checkpoint-bytes 1M`)

```
$ ./tarsnap -cvf a1 ~
... cancel once a checkpoint is written
$ ./tarsnap -cvf a1 ~
tarsnap: An archive already exists with the name "a1.part"
tarsnap: Error creating new archive
tarsnap: Error exit delayed from previous errors.
$ ./tarsnap --print-stats
                                       Total size  Compressed size
All archives                             22358736          1039660
  (unique data)                          12436176          1017924
$ ./tarsnap --fsck
...
$ ./tarsnap --print-stats
                                       Total size  Compressed size
All archives                             22359194          1040710
  (unique data)                          12436634          1018974
```

(the sizes increased after the `fsck`)

If we now delete `a1.part`, the size is 0 (as it should be).

**Comment by cperciva** [comment]
> illustrated without "recovered"

Eh?  Tarsnap is recovering a partial archive from a checkpoint in your example.

**Comment by gperciva** [comment]
I hate jargon.

**Comment by gperciva** [comment]
This was fixed in #219.
